backend/src/main/java/gov/cdc/usds/simplereport/config/authorization/UserAuthorizationVerifier.java

```diff
@@ -0,0 +1,33 @@
+package gov.cdc.usds.simplereport.config.authorization;
+
+import java.util.Optional;
+
+import gov.cdc.usds.simplereport.config.simplereport.AdminEmailList;
+import gov.cdc.usds.simplereport.service.OrganizationService;
+import gov.cdc.usds.simplereport.service.model.CurrentOrganizationRoles;
+import gov.cdc.usds.simplereport.service.model.IdentityAttributes;
+import gov.cdc.usds.simplereport.service.model.IdentitySupplier;
+
+public class UserAuthorizationVerifier {
+
+    private AdminEmailList _admins;
+    private IdentitySupplier _supplier;
+    private OrganizationService _orgService;
+
+    public UserAuthorizationVerifier(AdminEmailList admins, IdentitySupplier supplier, OrganizationService orgService) {
+        super();
+        this._admins = admins;
+        this._supplier = supplier;
+        this._orgService = orgService;
+    }
+
+    public boolean userHasSiteAdminRole() {
```

Comment:
this gets implicated in my above comment about generalizing the annotation

Reply:
Yeah—the best approach isn't available, but this layer we control both the object and the caller, so we can do better here. I think once we have the granular permissions this should be a single method that checks a specific permission. Doing that using SPeL is a bit of a funny-looking construction, but we can survive funny-looking.